packages/ts/lit-form/src/BinderNode.ts

```diff
@@ -52,17 +86,81 @@ export function getBinderNode<M extends AbstractModel>(model: M): BinderNode<M>
   return node as BinderNode<M>;
 }
 
+function getConstraintValidator<T>(constraint: Constraint<T>): Validator<T> {
+  const constraintTypes = [
+    Null,
+    NotNull,
+    AssertTrue,
+    AssertFalse,
+    Min,
+    Max,
+    DecimalMin,
+    Negative,
+    NegativeOrZero,
+    Positive,
+    PositiveOrZero,
+    Size,
+    Digits,
+    Past,
+    // PastOrPresent, // not supported
+    Future,
+    // FutureOrPresent, // not supported
+    Pattern,
+    NotEmpty,
+    NotBlank,
+    Email,
+  ];
+  for (const constraintType of constraintTypes) {
+    if (m.isConstraint(constraint, constraintType as NonAttributedConstraint)) {
+      // eslint-disable-next-line import/namespace
+      const Validator = Validators[constraintType.name] as Constructor<
+        Validator<T>,
+        EmptyObject extends typeof constraint.attributes
+          ? [typeof constraint.attributes | undefined]
+          : [typeof constraint.attributes]
+      >;
+      return new Validator(constraint.attributes);
+    }
+  }
+  throw new Error(`Unsupported constraint: ${constraint.name}`);
+}
+
+function getModelValidators<M extends ProvisionalModel>(model: M): ReadonlyArray<Validator<Value<M>>> {
+  if (model instanceof AbstractModel) {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    return model[_validators];
+  }
+
+  const validators: Array<Validator<Value<M>>> =
+    model instanceof NumberModel ? [new Validators.IsNumber(model[$optional]) as Validator<Value<M>>] : [];
+
+  for (const constraint of model[$constraints]) {
+    const validator = getConstraintValidator<Value<M>>(constraint);
+    validators.push(validator);
+  }
+
+  return validators;
+}
+
+function createEmptyValue<M extends ProvisionalModel>(model: M) {
+  if (model instanceof AbstractModel) {
+    return model.constructor.createEmptyValue();
+  }
+
+  return model[$defaultValue];
+}
```

Comment:
I guess, it'd need some docs to describe that there are some checks for transition period. Or, maybe, I'd put it in another file with all the proxies.